From 2af65b6e3651f21279fabf74643ab05b46c0cdf0 Mon Sep 17 00:00:00 2001
From: Fred Fan <r01011@freescale.com>
Date: Wed, 30 Jul 2008 15:39:45 +0800
Subject: [PATCH] ENGR00085518 i.MX51 Setup all the registers to allow userspace access

The registers of off-platform devices on AIPS#1&#2 can not be accessed
by memtool which an application runs in user space.
The soultion is:
    Permit the accessing from user space to them by setting AIPS OPACRs.

Signed-off-by: Fred Fan <r01011@freescale.com>
---
 arch/arm/mach-mx51/cpu.c |   24 ++++++++++++++++++++++++
 1 files changed, 24 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-mx51/cpu.c b/arch/arm/mach-mx51/cpu.c
index 824dbd2..516861c 100644
--- a/arch/arm/mach-mx51/cpu.c
+++ b/arch/arm/mach-mx51/cpu.c
@@ -34,3 +34,27 @@ void __init mxc_cpu_init(void)
 		mxc_set_system_rev(0x51, CHIP_REV_1_0);
 }
 
+static int __init post_cpu_init(void)
+{
+	unsigned int base, reg;
+
+	base = IO_ADDRESS(AIPS1_BASE_ADDR);
+	__raw_writel(0x0, base + 0x40);
+	__raw_writel(0x0, base + 0x44);
+	__raw_writel(0x0, base + 0x48);
+	__raw_writel(0x0, base + 0x4C);
+	reg = __raw_readl(base + 0x50) & 0x00FFFFFF;
+	__raw_writel(reg, base + 0x50);
+
+	base = IO_ADDRESS(AIPS2_BASE_ADDR);
+	__raw_writel(0x0, base + 0x40);
+	__raw_writel(0x0, base + 0x44);
+	__raw_writel(0x0, base + 0x48);
+	__raw_writel(0x0, base + 0x4C);
+	reg = __raw_readl(base + 0x50) & 0x00FFFFFF;
+	__raw_writel(reg, base + 0x50);
+
+	return 0;
+}
+
+postcore_initcall(post_cpu_init);
-- 
1.5.4.4

